toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -154,34 +169,60 @@ func updateGrubConfigForVerity(rootfsVerity imagecustomizerapi.Verity, rootHash
 	return nil
 }
 
-func constructVerityKernelCmdlineArgs(rootfsVerity imagecustomizerapi.Verity, rootHash string,
+func constructVerityKernelCmdlineArgs(verityInfo map[string]map[string]string,
 	partIdToPartUuid map[string]string, partitions []diskutils.PartitionInfo, buildDir string,
 ) ([]string, error) {
-	// Format the dataPartitionId and hashPartitionId using the helper function.
-	formattedDataPartition, err := systemdFormatPartitionId(rootfsVerity.DataDeviceId,
-		rootfsVerity.DataDeviceMountIdType, partIdToPartUuid, partitions, buildDir)
-	if err != nil {
-		return nil, err
-	}
+	var newArgs []string
+	addedVerityFlag := false
+
+	for verityType, verityData := range verityInfo {
+		var hashArg, dataArg, optionsArg, hashKey string
+
+		switch verityType {
+		case imagecustomizerapi.VerityRootDeviceName:
+			hashArg = "roothash"
+			dataArg = "systemd.verity_root_data"
+			hashKey = "systemd.verity_root_hash"
+			optionsArg = "systemd.verity_root_options"
+
+		case imagecustomizerapi.VerityUsrDeviceName:
+			hashArg = "usrhash"
+			dataArg = "systemd.verity_usr_data"
+			hashKey = "systemd.verity_usr_hash"
+			optionsArg = "systemd.verity_usr_options"
+
+		default:
+			return nil, fmt.Errorf("unsupported verity type: %s", verityType)
+		}
 
-	formattedHashPartition, err := systemdFormatPartitionId(rootfsVerity.HashDeviceId,
-		rootfsVerity.HashDeviceMountIdType, partIdToPartUuid, partitions, buildDir)
-	if err != nil {
-		return nil, err
-	}
+		formattedDataPartition, err := systemdFormatPartitionId(verityType,
```

Comment:
The `verityInfo` contains the partition UUID for the data partition and hash partition. So, you can refactor the `systemdFormatPartitionId` function to take the partition UUID directly instead of the partition ID and `partIdToPartUuid` map.